src/status_im/protocol/handlers.cljs

```diff
@@ -353,15 +358,22 @@
             (when route-event {:dispatch route-event})))))))
 
 (defn update-message-status [db {:keys [message-id ack-of-message group-id from status]}]
-  (let [message-id' (or ack-of-message message-id)
-        group?      (boolean group-id)
-        status-path (if (and group? (not= status :sent))
-                      [:message-data :user-statuses message-id' from]
-                      [:message-data :statuses message-id'])
-        {current-status :status} (get-in db status-path)]
-    (if-not (= :seen current-status)
-      (assoc-in db status-path {:whisper-identity from
-                                :status           status})
+  (let [message-id'          (or ack-of-message message-id) 
+        update-group-status? (and group-id (not= status :sent))
+        message-path         (if update-group-status?
+                               [:chats group-id :messages message-id']
+                               [:chats (or group-id from) :messages message-id'])
```

Comment:
it looks like we don't need `if` here, it can be just `[:chats (or group-id from) :messages message-id']` for both cases, no?

Reply:
Unfortunately not, because even when `group-id` is not nil, but `update-group-status?` is true (so the case when it's the group message, but it's some other status then `:sent`) being updated, we have to ignore group id...
The logic is super bloated and hard to read, we don't even need both `:user-statuses` and `:message-status` in message data object (and realm), `:user-statuses` us totally sufficient and would result in more straightforward code and easier to grasp logic.
I planned to tackle that in next refactoring, but if you wish, I can do it right away in this PR !

Reply:
> we have to ignore group id...
but... we don't ignore it now, if it is not nil it will be alway used atm. Or i'm missing something?

Reply:
You are right, for group message, we always have to look-it up using group-id, will fix it